recipes/openssl/3.x.x/conanfile.py

```diff
@@ -0,0 +1,630 @@
+import os
+import fnmatch
+import platform
+from conans.errors import ConanInvalidConfiguration, ConanException
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.tools import Version
+
+
+class OpenSSLConan(ConanFile):
+    name = "openssl"
+    settings = "os", "compiler", "arch", "build_type"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openssl/openssl"
+    license = "Apache-2.0"
+    topics = ("conan", "openssl", "ssl", "tls", "encryption", "security")
+    description = "A toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "386": [True, False],
+        "capieng_dialog": [True, False],
+        "enable_capieng": [True, False],
+        "no_asm": [True, False],
+        "no_async": [True, False],
+        "no_bf": [True, False],
+        "no_cast": [True, False],
+        "no_deprected": [True, False],
+        "no_des": [True, False],
+        "no_dh": [True, False],
+        "no_dsa": [True, False],
+        "no_dso": [True, False],
+        "no_engine": [True, False],
+        "no_fips": [True, False],
+        "no_hmac": [True, False],
+        "no_legacy": [True, False],
+        "no_md5": [True, False],
+        "no_mdc2": [True, False],
+        "no_rc2": [True, False],
+        "no_rc4": [True, False],
+        "no_rsa": [True, False],
+        "no_sha": [True, False],
+        "no_sse2": [True, False],
+        "no_threads": [True, False],
+        "openssldir": "ANY",
+        "zlib": [True, False]
```

Comment:
This option was named `no_zlib` in 1.x.
If you intended to rename it, maybe go for `with_zlib`?

Reply:
Also, in 1.1 this option was removed.
Has it come back? Was it ever removed?

Reply:
It looks to me like the upstream option `no_zlib` was last available in 1.0.1 https://github.com/openssl/openssl/blob/OpenSSL_1_0_1g/INSTALL#L60

then got removed in >= 1.1.1

However, a `zlib` option was always available

https://github.com/openssl/openssl/blob/OpenSSL_1_1_1h/INSTALL#L550
https://github.com/openssl/openssl/blob/master/INSTALL.md#zlib

That is where the name is coming from. I followed upstream